### PR TITLE
Follow-up for RetryQueryRunner fix

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

Addresses the [last comment](https://github.com/apache/druid/pull/10082#issuecomment-652576873) of https://github.com/apache/druid/pull/10082 except integration tests. Will add them in a separate PR. This PR also modifies how to check the `shouldFailOnTruncatedResponseContext` flag; now query servers and the broker use different config initialized by guice. I think this way is more stable and less error-prone than using query context.

<hr>

This PR has:
- [x] been self-reviewed.
   - [ ] using the [concurrency checklist](https://github.com/apache/druid/blob/master/dev/code-review/concurrency.md) (Remove this item if the PR doesn't have any relation to concurrency.)
- [x] added documentation for new or modified features or behaviors.
- [ ] added Javadocs for most classes and all non-trivial methods. Linked related entities via Javadoc links.
- [ ] added or updated version, license, or notice information in [licenses.yaml](https://github.com/apache/druid/blob/master/licenses.yaml)
- [ ] added comments explaining the "why" and the intent of the code wherever would not be obvious for an unfamiliar reader.
- [ ] added unit tests or modified existing tests to cover new code paths, ensuring the threshold for [code coverage](https://github.com/apache/druid/blob/master/dev/code-review/code-coverage.md) is met.
- [ ] added integration tests.
- [x] been tested in a test Druid cluster.

### CODE REVIEW

**Comment by jihoonson** [comment]
The test coverage check for `ReportTimelineMissingSegmentQueryRunner` is a false alarm since I only added a logging.

File: services/src/main/java/org/apache/druid/cli/CliBroker.java [hunk 1]
```diff
@@ -102,6 +103,12 @@ public CliBroker()
           binder.bindConstant().annotatedWith(Names.named("tlsServicePort")).to(8282);
           binder.bindConstant().annotatedWith(PruneLoadSpec.class).to(true);
           binder.bindConstant().annotatedWith(PruneLastCompactionState.class).to(true);
+          binder.bindConstant()
```
Comment (gianm) [inline services/src/main/java/org/apache/druid/cli/CliBroker.java:106]: It looks like this should work, but why not create a reasonable ResponseContextConfig and bind it directly to `ResponseContextConfig.class`? Seems like that would be simpler.

File: docs/querying/querying.md [hunk 2]
```diff
@@ -139,4 +139,5 @@ Possible codes for the *error* field include:
...
 |`Unsupported operation`|The query attempted to perform an unsupported operation. This may occur when using undocumented features or when using an incompletely implemented extension.|
+|`Truncated response context`|The response context is an internal data structure to store intermediate state during query processing. Some of states are collected from query servers (such as historicals or realtime tasks) and merged in the Broker. To send those states to the Broker, the query servers serialize them in the HTTP header and send it together with the stream of intermediate query result. As the max size of an HTTP header is limited to 7 KB in Druid, this error can occur if JSON-serialized response context is larger than the limit.<br/><br/>This error shouldn't happen in most cases. A possible cause of this error can be too many segments moving (for segment balancing) at the time when a query is issued which reads lots of segments. You should be able to see debug logs about missing segments in the historical logs if you are facing this case.|
```
[the remark this answers is not in the document]
Comment (jihoonson) [reply docs/querying/querying.md]: 👍

File: services/src/main/java/org/apache/druid/cli/CliBroker.java (hunk 1 above)
Comment (jihoonson) [reply services/src/main/java/org/apache/druid/cli/CliBroker.java]: Ah, there was only one constant to bind at first (`shouldFailOnTruncatedResponseContext`), and so I just added a constant binding. Later, to make the test coverage check happier, I added another constant `maxResponseContextHeaderSize` and just copy-pasted existing code 😅. Changed to directly bind the context class.

**Comment by jihoonson** [comment]
```
33  F | L           private static final Logger LOG = new Logger(ReportTimelineMissingSegmentQueryRunner.class);
34  F             
45  F | L             LOG.debug("Reporting a missing segment[%s] for query[%s]", descriptor, queryPlus.getQuery().getId());
------------------------------------------------------------------------------
Diff coverage statistics:
------------------------------------------------------------------------------
|     lines      |    branches    |   functions    |   path
------------------------------------------------------------------------------
|   0% (0/2)     | 100% (0/0)     | 100% (3/3)     | org/apache/druid/query/ReportTimelineMissingSegmentQueryRunner.java
------------------------------------------------------------------------------
Total diff coverage:
 - lines: 0% (0/2)
 - branches: 100% (0/0)
 - functions: 100% (3/3)
ERROR: Insufficient line coverage of 0% (0/2). Required 50%.
```

@gianm @clintropolis thank you for the review. I fixed the spell check failure, but the test coverage check is just not happy about the logging I added. I'm going to merge this PR.

**Pull request merged**